### DISCUSSION TITLE
SSL keystore/truststore configuration

### DISCUSSION THREAD

**Discussion by kishore-pothireddy** [dropwizard/dropwizard-kafka#203] [Q&A] [answered]
Hi,

How can we add following SSL keystore/truststore configuration to kafka producer? I don't see any of these properties declared in securityFactory.
```text
security.protocol=SSL
ssl.truststore.location=../ssl/client.truststore.jks
ssl.truststore.password=****
ssl.keystore.location=../ssl/client.kesytore.jks
ssl.keystore.password=****
```

**Comment by pstackle** [comment, marked as answer]
I agree, I don't see the bottom 4 config values in either `SslSecurityFactory` or `SecurityFactory`. The first, `security.protocol` is covered by using the  `SslSecurityFactory` via the `securityProtocol` property. 

So, even though the properties you want aren't explicitly defined on a security factory, you should still be able to set those values by using the bundle constructor that takes two parameters, where the second is: `Map<String, Object> configOverrides`.

The `configOverrides` values get merged in with all the other config values that are generated from the various config factories, so the ultimately configured/built kafka client will receive those key/value combinations.
